api/views.py

```diff
@@ -1,10 +1,25 @@
-from api.serializers import UserSerializer
-from rest_framework import viewsets
+from api.models import Todo
+from api.serializers import TodoSerializer, UserSerializer
+from rest_framework import viewsets, permissions, renderers
 from django.contrib.auth.models import User
+from api.permissions import IsAuthorOrReadOnly
+from rest_framework.decorators import action
 
 
 class UserViewSet(viewsets.ReadOnlyModelViewSet):
 
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
+
+class TodoViewSet(viewsets.ModelViewSet):
+   
+    queryset = Todo.objects.all()
+    serializer_class = TodoSerializer
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly,
```

Comment:
You should not allow reading from the Todo endpoint if user is not authenticated at all. Replace `permissions.IsAuthenticatedOrReadOnly` with `permissions.IsAuthenticated`. I like the use of `IsAuthorOrReadOnly` so leave it as it is.